### ISSUE TITLE
unable to recompile SMusic.apk on apktool 2.0.2

### ISSUE THREAD

**Issue by JoeyDuran** [iBotPeaches/Apktool#1117]
i can decompile just fine, but i am unable to recompile the apk, even without changes
Note 5 running Lollipop 5.1.1

**Comment by iBotPeaches** [comment]
I need an apk or a stacktrace to work with here.

**Comment by JoeyDuran** [comment]
sorry about that
[SMusic.zip](https://github.com/iBotPeaches/Apktool/files/72775/SMusic.zip)

**Comment by iBotPeaches** [comment]
```
Error: No resource found that matches the given name (at 'style' with value '@style/Text.SansSerif.Marquee').

Error: No resource found that matches the given name (at 'style' with value '@style/Text.SansSerif.Marquee.MiniPlayer').

Error: No resource found that matches the given name (at 'style' with value '@style/Text.Widget.Marquee').

No resource found that matches the given name (at 'style' with value '@style/SeekBar.mini_player_progressbar').

Error: No resource found that matches the given name (at 'style' with value '@style/Text.Widget.Marquee').
```

5 missing resources. There might be broken resources in the `res/values/styles.xml`.

```
ibotpeaches@tombstone:~/Downloads/Apktool/Bug1117$ cat SMusic/res/values/styles.xml | grep 'APKTOOL'
    <item name="android:src">@drawable/APKTOOL_DUMMY_0082</item>
    <item type="style" name="APKTOOL_DUMMY_0087">false</item>
    <item type="style" name="APKTOOL_DUMMY_0088">false</item>
    <item type="style" name="APKTOOL_DUMMY_0089">false</item>
    <item type="style" name="APKTOOL_DUMMY_008a">false</item>
```

Apktool has no fix for this currently. Resources are referenced that cannot be found, thus dummy resources which don't match the naming of the other resources.

We might need to follow resources to identify if they exist, if not name the dummy resource as such vs `APKTOOL_DUMMY`. However, there is a reason we currently don't do that. Once we are writing files out, the decode of resources is done. The structure of how apktool operates would need to change. 

This is a duplicate of something here, but I wrote more of a proper response in this report. So will keep this one around.

**Comment by iBotPeaches** [comment]
```
➜  Bug1117 aapt d resources SMusic.apk | grep -r -i 'TYPE CONFIG'
grep: warning: recursive search of stdin
(standard input):      INVALID TYPE CONFIG FOR RESOURCE 0x7f020050
(standard input):      INVALID TYPE CONFIG FOR RESOURCE 0x7f020082
(standard input):      INVALID TYPE CONFIG FOR RESOURCE 0x7f0200ca
(standard input):      INVALID TYPE CONFIG FOR RESOURCE 0x7f0200cb
(standard input):      INVALID TYPE CONFIG FOR RESOURCE 0x7f0200ce
(standard input):      INVALID TYPE CONFIG FOR RESOURCE 0x7f0b00bc
(standard input):      INVALID TYPE CONFIG FOR RESOURCE 0x7f0c0229
(standard input):      INVALID TYPE CONFIG FOR RESOURCE 0x7f0c022a
(standard input):      INVALID TYPE CONFIG FOR RESOURCE 0x7f110086
(standard input):      INVALID TYPE CONFIG FOR RESOURCE 0x7f110087
(standard input):      INVALID TYPE CONFIG FOR RESOURCE 0x7f110088
(standard input):      INVALID TYPE CONFIG FOR RESOURCE 0x7f110089
(standard input):      INVALID TYPE CONFIG FOR RESOURCE 0x7f11008a
```

Nothing we can do at this point. Apktool can't determine the type of the resource, so it can't reliably create a fake Style for these resources. Manual intervention to recreate the styles and/or remove them is the only solution.

**Comment by Lanchon** [comment]
> Manual intervention to recreate the styles

can you point to a discussion regarding this? thanks!
